### ISSUE TITLE
designspace axis map must not contain same keys mapping to different values

### ISSUE THREAD

**Issue by anthrotype** [googlei18n/glyphsLib#341]
it's called "map" because it works like a mapping of keys (user locations) to values (design space locations). Just like in python dicts, one key cannot map to multiple values.

this is how varLib is translating from the "input" and "output" attributes of map elements to a python dictionary:
https://github.com/fonttools/fonttools/blob/a523a697164a4cdc721a06dec80791d4b1496f12/Lib/fontTools/varLib/__init__.py#L626

So if more than one "input" with same value is present, the last one wins and overwrites the previous ones.

Now, because of the way we map Thin, ExtraLight and UltraLight all to 250 weight class (see classes.py WEIGHT_CODES), we end up writing things like this (this is GlyphsUnitTestSans.glyphs, exported to designspace):

```xml
<?xml version='1.0' encoding='utf-8'?>
<designspace format="3">
    <axes>
        <axis default="400" maximum="900" minimum="250" name="Weight" tag="wght">
            <labelname xml:lang="en">Weight</labelname>
            <map input="250" output="17" />
            <map input="250" output="30" />
            <map input="300" output="55" />
            <map input="400" output="75" />
            <map input="400" output="90" />
```

In this font, there are two Thin and Extra Light instances that are assigned a 250 weight class. And there is even a "Web" instance that has default 400 weight class but has a different design location (75) than the Regular instance. In the latter case, one would expect the user to add a weightClass custom parameter (e.g. 350, in between Light and Regular) to make this unique and avoid the clash.
 
Note that this test font does not use the new Axes and Axis Location custom parameters, therefore is using the user location of the instances to make the axis map elements.

Even if the Axis Location was used we should raise an error if two masters have the same Axis Location but different design space locations, to prevent such invalid map elements.

**Comment by anthrotype** [comment]
there may be good legacy reasons (e.g. support some old MS Word) to have OS/2 weight class set to 250 for thin and extra light styles, but for the designspace axis maps -- which are primarily meant to be used to build a non-linear normalization mapping in the avar table -- we don't want that to happen.
The user location of a Thin named instance in a variable font is 100, not 250.

I just compared opening GlyphsUnitTestSans.glyphs in latest stable Glyphs.app (2.4.4 1075) and in latest beta (2.5b 1126) and it appears that in the latest beta, the default weight class for Thin and Extra Light is respectively 100 and 200, instead of the legacy 250 value.

So in addition of asserting that same input map to same output values (in axes.to_designspace_axes), we should also modify the default WEIGHT_CODE in classes.py to match the latest Glyphs.app beta.

**Comment by anthrotype** [comment]
and if one really wants to have a Thin or Extra Light instance's weight class be 250, one can always use the weightClass custom parameter.

Although again if more than one instance have the same weight class, we face again the same issue when constructing the axis mappings from the instances user locations...

**Comment by anthrotype** [comment]
fixed by #343
